javatools/build.gradle.kts

```diff
@@ -98,6 +101,23 @@ val jar by tasks.existing(Jar::class) {
     }
 }
 
+/**
+ * TODO: Marcus suggests that merging the jars is not a good way to go. Instead we simply need to
+ *       get the jline.jar into our distribution (installDist task) and point to it in the manifest
+ *       of the javatools.jar
+ */
+val shadowJar by tasks.existing(ShadowJar::class) {
+    dependsOn(jar)
+
+    archiveBaseName = "javatools"
+
+    from(tasks.jar.get().archiveFile)
```

Comment:
This forces the existing javatools fat jar, that master uses, to get executed archived, and put together just by configuring the build. Everything not in a doLast or doFirst block will run at configuration time, so this means that if gradle just starts looking at the build, this from statement forces it to build the non shadow jar. That is probably not what you want.  Dependent tasks should be resolved lazy and as late as possible. 

The tasks.jar is a task container, i.e. provider of a task. By calling get outside a doLast and doFirst block you force Gradle to compute that task as soon at is sees this statement by scanning the code. 

You are also asking for an output of the task, the archiveFile (javatools-0.4.43.jar) in the configuration phase after the get, so you can be sure just loading this file with gradle will force build javatools.

Reply:
I would really appreciate if you fixed it to do what you suggested earlier.